src/function/matrix/map.js

```diff
@@ -9,27 +10,36 @@ export const createMap = /* #__PURE__ */ factory(name, dependencies, ({ typed })
    * Create a new matrix or array with the results of a callback function executed on
    * each entry of a given matrix/array.
    *
-   * For each entry of the input, the callback is invoked with three arguments:
-   * the value of the entry, the index at which that entry occurs, and the full
-   * matrix/array being traversed. Note that because the matrix/array might be
+   * For each entry of the input,
+   *
+   * the callback is invoked with 2N + 1 arguments:
+   * the N values of the entry, the index at which that entry occurs, and the N full
+   * broadcasted matrix/array being traversed where N is the number of matrices being traversed.
+   * Note that because the matrix/array might be
    * multidimensional, the "index" argument is always an array of numbers giving
    * the index in each dimension. This is true even for vectors: the "index"
    * argument is an array of length 1, rather than simply a number.
    *
    * Syntax:
    *
    *    math.map(x, callback)
+   *    math.map(x, y, callback)
```

Comment:
How about documenting the syntax like:

```
math.map(x, callback)
math.map(x, y, ..., callback)
```

? 

Then it is clear (I think) that you can also pass more than two arrays/matrices.